src/App.tsx

```diff
@@ -1,8 +1,113 @@
-import React from 'react';
+import React, { useState } from 'react';
+import TodosList from './TodosList';
 import './App.css';
+import { promise } from './API';
 
-const App = () => (
-  <h1>Dynamic list of TODOs</h1>
-);
+const App = () => {
+  const [todos, setTodos] = useState<TodosWithUsers[]>([]);
+  const [isLoading, setLoading] = useState<boolean>(false);
+  const [upAlphabetName, setUpAlphabetName] = useState<boolean>(false);
+  const [upAlphabetTitle, setUpAlphabetTitle] = useState<boolean>(false);
+  const [isCompletedTodo, setIsCompletedTodo] = useState<boolean>(false)
+
+  const fetchData = (): void => {
+    setLoading(true);
+    promise.then(data => {
+      setLoading(false);
+
+      return setTodos(data);
+    });
+  };
+
+  const nameFilter = () => {
+    let newTodos = [];
+
+    if (!upAlphabetName) {
+      newTodos = todos.sort((a, b) => a.user.name.localeCompare(b.user.name));
+      setUpAlphabetName(true);
```

Comment:
you say `set up alphabet name` but store boolean value, why? if you use such name the value should be a name or just change the name to the appropriate one

+ what is alphabet name?

Reply:
no fixes and no comments...